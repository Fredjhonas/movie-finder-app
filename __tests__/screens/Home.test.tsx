/**
 * @format
 */

import 'react-native';

// Note: import explicitly to use the types shipped with jest.
import { expect, it, jest } from '@jest/globals';
import React from 'react';
// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

import HomeScreen from '../../src/screens/home';

jest.mock('native-base', () => {
  return {
    Center: 'Center',
  };
});

it('renders correctly', () => {
  renderer.create(<HomeScreen />);
});

it('renders correctly with searchbar open', () => {
  jest.mock('../../src/hooks/useSearchbar', () => {
    return {
      useSearchbar: () => ({ isOpen: true }),
    };
  });
  const tree = renderer.create(<HomeScreen />);
  expect(tree.toJSON()).toMatchSnapshot();
});

it('renders correctly with searchbar closed', () => {
  jest.mock('../../src/hooks/useSearchbar', () => {
    return {
      useSearchbar: () => ({ isOpen: false }),
    };
  });
  const tree = renderer.create(<HomeScreen />);
  expect(tree.toJSON()).toMatchSnapshot();
});
