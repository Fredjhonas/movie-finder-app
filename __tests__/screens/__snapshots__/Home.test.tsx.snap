// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly with searchbar closed 1`] = `<Center />`;

exports[`renders correctly with searchbar open 1`] = `<Center />`;
